### DISCUSSION TITLE
Need help explaining SHAP dependence_plot

### DISCUSSION THREAD

**Discussion by arilwan** [shap/shap#3067] [Q&A]
Guys,

I am in need of help interpreting SHAP dependence_plot of my model. It is a random forest classification model involving 5 classes `(20206 samples, 24 features)`, so I used `shap.TreeExplainer()`  to explain like so:
```python
explainer = shap.TreeExplainer(model)
shap_values = explainer.shap_values(X_test)

# saving shap values -- this is a list data xture
with open('shap_values.h5', 'wb') as file:
	joblib.dump(shap_values, file)

np.save('shap_values.npy', shap_values) # shap values saved as ndarray
```
Later I loaded this values for analysis. 
```python
with open('shap_values.h5', 'rb') as file:
	shap_values_list = joblib.load(file)

shap_values_arr = np.load('shap_values.npy')
shap_values_arr.shape
(5, 20206, 24)
```
So I understand correctly, my `shap_values_list` would have a list of arrays, each corresponding to each class's SHAP values per feature. I assumed the first enrty to  `shap_values_list` corresponds to class `0`, the second to class `1`etc.

My `summary_plot` shows `speed_max` is the most dorminant feature, so I want to visualise its interaction with other feature for each class. Here I produce the `dependence_plot` for class `0` (walk class) and class `3` (car class).
Class `0` (walk):

![Screenshot 2023-07-06 at 12 16 32](https://github.com/slundberg/shap/assets/42409395/0c896c3e-9fa7-49bc-b058-5dc89361f1ed)

Class `3` (car):

![Screenshot 2023-07-06 at 12 16 57](https://github.com/slundberg/shap/assets/42409395/b28ecde7-3fd1-4fb8-9e09-97317acc360c)

What I find strange, and cannot really understand why, is the range of values of `speed_max` (horizontal axis). I am pretty sure, the highest `speed_max` value for class `0` is `9.8`.  Of couse the car class `3`  has a near 50 value for that feature `48.64`. 

Why is the range of `speed_max` value for each class is `0-50` (the same applies to bicycle class `1` having `speed_max` highest value of `12.24`?

Can anyone please help explain this to me?

**Comment by thatlittleboy** [comment]
`shap_values[0]` (class==walk) shape is `(20206, 24)`, i.e., it contains all of your dataset points (inclusive of those whose true label is car, bike, etc.). The feature values (`speed_max` and `speed_p25` here) are two columns from that array, so it shouldn't be surprising to see the full range of values in all of your plots.
